### PR TITLE
Restructure and Support for plain status list

### LINKED ISSUE #41: Does Status List need to be signed as JWT?
@bc-pi questions whether Status List needs to be signed as a JWT and what are the arguments for doing so

### LINKED ISSUE #55: Define a protocol how to fetch the status list token
Using HTTP GET...

### LINKED ISSUE #56: Add option for plain status list (unsigned)

### PR DESCRIPTION
**Pull request by paulbastian**
Closes #55
Closes #56
Closes #41

## 📑 Description
Adds support for unsigned Status List retrieved via HTTP endpoint.
Also restructures the document to allow this.

Remaining tasks:

- [x] Fix python example to provide Status List Object in 4.1
- [x] Fix python example in 5.1 to use the very same Status List Object from 4.1

## Preview Link

[click here for rendered preview of PR](https://vcstuff.github.io/draft-ietf-oauth-status-list/56-add-option-for-plain-status-list-unsigned/draft-ietf-oauth-status-list.html)

### CODE REVIEW

**Review by c2bo — APPROVED** [review]

File: draft-ietf-oauth-status-list.md [hunk 1]
```diff
@@ -81,69 +89,137 @@ Relying Party:
...
 
 Referenced Token:
-: A token in JWT or CWT representation which contains a reference to a Status List Token. The information from the contained Status List may give a Relying Party additional information about up-to-date status of the Referenced Token.
+: A token in JWT or CWT representation which contains a reference to a Status List or Status List Token. The information from the contained Status List may give a Relying Party additional information about up-to-date status of the Referenced Token.
 
-# JSON Web Token Representation
+# Status List {#status-list}
 
-## Status List JWT Format and Processing Requirements {#jwt-status-list-format-and-processing}
+A Status List is a byte array that contains the statuses of many Referenced Tokens represented by one or multiple bits. A common representation of a Status List is composed by the following algorithm:
```
Comment (tplooker) [inline draft-ietf-oauth-status-list.md:102]: With the proposal of #90 we would need to re-word this to be a bit more abstract right? Given the status list might take a different form?

File: draft-ietf-oauth-status-list.md [hunk 2]
```diff
@@ -86,69 +94,137 @@ Relying Party:
...
+* `iss`: REQUIRED. The `iss` (issuer) claim MUST specify a unique string identifier for the entity that issued the Status List Token. In the absence of an application profile specifying otherwise, compliant applications MUST compare issuer values using the Simple String Comparison method defined in Section 6.2.1 of {{RFC3986}}. The value MUST be equal to that of the `iss` claim contained within the Referenced Token.
+* `sub`: REQUIRED. The `sub` (subject) claim MUST specify a unique string identifier for that Status List Token. The value MUST be equal to that of the `uri` claim contained in the `status` claim of the Referenced Token.
+* `iat`: REQUIRED. The `iat` (issued at) claim MUST specify the time at which the Status List Token was issued.
+* `exp`: OPTIONAL. The `exp` (expiration time) claim MAY convey the time at which it is considered expired by its issuer.
+* `status_list`: REQUIRED. The `status_list` (status list) claim MUST specify the Status List conforming to the rules outlined in [](#status-list-json).
 
-## Referenced Token Format and Processing Requirements {#jwt-referenced-token}
+Further rules apply:
```
Comment (tplooker) [inline draft-ietf-oauth-status-list.md:192]:
```suggestion
The following additional rules apply:
```

File: draft-ietf-oauth-status-list.md [hunk 3]
```diff
@@ -86,69 +94,137 @@ Relying Party:
...
+The following content applies to the JWT Claims Set:
 
-3. The result of the compression is then base64url-encoded, as defined in Section 2 of {{RFC7515}}.
+* `iss`: REQUIRED. The `iss` (issuer) claim MUST specify a unique string identifier for the entity that issued the Status List Token. In the absence of an application profile specifying otherwise, compliant applications MUST compare issuer values using the Simple String Comparison method defined in Section 6.2.1 of {{RFC3986}}. The value MUST be equal to that of the `iss` claim contained within the Referenced Token.
+* `sub`: REQUIRED. The `sub` (subject) claim MUST specify a unique string identifier for that Status List Token. The value MUST be equal to that of the `uri` claim contained in the `status` claim of the Referenced Token.
+* `iat`: REQUIRED. The `iat` (issued at) claim MUST specify the time at which the Status List Token was issued.
+* `exp`: OPTIONAL. The `exp` (expiration time) claim MAY convey the time at which it is considered expired by its issuer.
+* `status_list`: REQUIRED. The `status_list` (status list) claim MUST specify the Status List conforming to the rules outlined in [](#status-list-json).
```
Comment (tplooker) [inline draft-ietf-oauth-status-list.md:190]: Im unsure whether this layout of a bullet list of claims is any clearer then the number bullet points we had previously, any particular reason for this change?

Comment (paulbastian) [reply draft-ietf-oauth-status-list.md:190]: it allows to describe nested claims and shows more clearly the structure.
Therefore its more dense making the document shorter and better to read imo

Comment (paulbastian) [reply draft-ietf-oauth-status-list.md:190]: also it makes things more obvious when REQUIRED/OPTIONAL is the first word in the description

**Review by tplooker — APPROVED** [review]

**Pull request merged**